pipeline/src/lib.rs

```diff
@@ -47,6 +48,37 @@ pub fn access_element<T: FieldElement>(
     }
 }
 
+#[allow(clippy::print_stdout)]
+pub fn serde_data_to_query_callback<T: FieldElement, S: serde::Serialize + Send + Sync>(
+    channel: u32,
+    data: &S
+) -> impl QueryCallback<T> {
+    let bytes = serde_json::to_vec(&data).unwrap();
+    move |query: &str| -> Result<Option<T>, String> {
+        match &parse_query(query)?[..] {
+            ["\"data\"", index, what] => {
+                let what = what
+                    .parse::<u32>()
+                    .map_err(|e| format!("Error parsing what: {e})"))?;
+
+                if what != channel {
```

Comment:
Why don't we support multiple channels?

Reply:
This is the specific callback created for a single channel. The complete callback in Pipeline is actually a sequence of nested callbacks, where if the outer most fails (returns None) it tries the next inner one.